### PR TITLE
Add support for azure china

### PR DESCRIPTION
**Pull request by chaosddp**
## What changes are proposed in this pull request?

Update the regex pattern to parse azure wasbs uri for azure china.

## How is this patch tested?

Unit tests (tests/store/artifact/test_azure_blob_artifact_repo.py)

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/artifact/azure_blob_artifact_repo.py [hunk 1]
```diff
@@ -60,20 +60,21 @@ def parse_wasbs_uri(uri):
         parsed = urllib.parse.urlparse(uri)
         if parsed.scheme != "wasbs":
             raise Exception("Not a WASBS URI: %s" % uri)
-        match = re.match("([^@]+)@([^.]+)\\.blob\\.core\\.windows\\.net", parsed.netloc)
+        match = re.match("([^@]+)@([^.]+)\\.(blob\\.core\\.(windows\\.net|chinacloudapi\\.cn))", parsed.netloc)
```
Comment (harupy) [inline mlflow/store/artifact/azure_blob_artifact_repo.py:63]: Can we refactor this line to use a raw string?

```suggestion
        match = re.match(r"([^@]+)@([^.]+)\.(blob\.core\.(windows\.net|chinacloudapi\.cn))", parsed.netloc)
```

Comment (harupy) [reply mlflow/store/artifact/azure_blob_artifact_repo.py:63]: Can we also fix the docstring of this function?

**Comment by harupy** [comment]
@chaosddp The proposed changes make sense to me! Could you fix the DCO and lint check failures?

File: mlflow/store/artifact/azure_blob_artifact_repo.py [hunk 2]
```diff
@@ -60,20 +60,21 @@ def parse_wasbs_uri(uri):
...
         if match is None:
             raise Exception(
-                "WASBS URI must be of the form " "<container>@<account>.blob.core.windows.net"
+                "WASBS URI must be of the form " "<container>@<account>.blob.core.windows.net" " or <container>@<account>.blob.core.chinacloudapi.cn"
             )
         container = match.group(1)
         storage_account = match.group(2)
+        api_uri = match.group(3)
```
Comment (harupy) [inline mlflow/store/artifact/azure_blob_artifact_repo.py:70]:
```suggestion
        api_uri_suffix = match.group(3)
```

Can we add `_suffix` here?

File: mlflow/store/artifact/azure_blob_artifact_repo.py (hunk 1 above)
Comment (chaosddp) [reply mlflow/store/artifact/azure_blob_artifact_repo.py]: fixed.

File: mlflow/store/artifact/azure_blob_artifact_repo.py (hunk 2 above)
Comment (chaosddp) [reply mlflow/store/artifact/azure_blob_artifact_repo.py]: added.

File: mlflow/store/artifact/azure_blob_artifact_repo.py [hunk 3]
```diff
@@ -49,31 +49,32 @@ def __init__(self, artifact_uri, client=None):
...
+        """Parse a wasbs:// URI, returning (container, storage_account, path, api_uri_suffix)."""
         parsed = urllib.parse.urlparse(uri)
         if parsed.scheme != "wasbs":
             raise Exception("Not a WASBS URI: %s" % uri)
-        match = re.match("([^@]+)@([^.]+)\\.blob\\.core\\.windows\\.net", parsed.netloc)
+        match = re.match(r"([^@]+)@([^.]+)\.(blob\.core\.(windows\.net|chinacloudapi\.cn))", parsed.netloc)
         if match is None:
             raise Exception(
-                "WASBS URI must be of the form " "<container>@<account>.blob.core.windows.net"
+                "WASBS URI must be of the form " "<container>@<account>.blob.core.windows.net" " or <container>@<account>.blob.core.chinacloudapi.cn"
```
Comment (harupy) [inline mlflow/store/artifact/azure_blob_artifact_repo.py:66]:
```suggestion
                "WASBS URI must be of the form <container>@<account>.blob.core.windows.net"
                " or <container>@<account>.blob.core.chinacloudapi.cn"
```

Can we fix this line?

**Comment by harupy** [comment]
@chaosddp Can you fix the lint failure: https://github.com/mlflow/mlflow/runs/4319606424?check_suite_focus=true?

File: mlflow/store/artifact/azure_blob_artifact_repo.py (hunk 3 above)
Comment (chaosddp) [reply mlflow/store/artifact/azure_blob_artifact_repo.py]: fixed, not sure why pylint not work locally.

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by chaosddp** [comment]
Lint for test, sorry, will update it...

**Comment by chaosddp** [comment]
> Lint for test, sorry, will update it...

**Pull request merged**